anxcloud/data_source_ip_address_test.go

```diff
@@ -0,0 +1,164 @@
+					resource.TestCheckResourceAttrPair("data.anxcloud_ip_address.test_id", "vlan_id", "anxcloud_ip_address.test_v4_x_x_x_5", "vlan_id"),
+				),
+			},
+
+			// Get IPv4 address by it's "address"
+			{
+				Config: testResources + `
+				data "anxcloud_ip_address" "test_address" {
+					address    = anxcloud_ip_address.test_v4_x_x_x_5.address
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.anxcloud_ip_address.test_address", "vlan_id", "anxcloud_ip_address.test_v4_x_x_x_5", "vlan_id"),
+				),
+			},
+
+			// Get another IPv4 address with same address-name-prefix as previous (to test search-match loop)
+			{
+				Config: testResources + `
+				data "anxcloud_ip_address" "test_address" {
+					address    = anxcloud_ip_address.test_v4_x_x_x_50.address
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.anxcloud_ip_address.test_address", "id", "anxcloud_ip_address.test_v4_x_x_x_50", "id"),
+				),
+			},
+
+			// Get IPv4 address by it's "address" and network_prefix_id
+			{
+				Config: testResources + `
+				data "anxcloud_ip_address" "test_address" {
+					address           = anxcloud_ip_address.test_v4_x_x_x_5.address
+					network_prefix_id = anxcloud_ip_address.test_v4_x_x_x_5.network_prefix_id
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.anxcloud_ip_address.test_address", "vlan_id", "anxcloud_ip_address.test_v4_x_x_x_5", "vlan_id"),
+				),
+			},
+
+			// Get IPv4 address by it's "address" and vlan_id
+			{
+				Config: testResources + `
+				data "anxcloud_ip_address" "test_address" {
+					address = anxcloud_ip_address.test_v4_x_x_x_5.address
+					vlan_id = anxcloud_ip_address.test_v4_x_x_x_5.vlan_id
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.anxcloud_ip_address.test_address", "vlan_id", "anxcloud_ip_address.test_v4_x_x_x_5", "vlan_id"),
+				),
+			},
+
+			// Get IPv6 address by it's shortened "address"
+			{
+				Config: testResources + `
+				data "anxcloud_ip_address" "test_v6_short" {
+					address = anxcloud_ip_address.test_v6.address
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.anxcloud_ip_address.test_v6_short", "network_prefix_id", "anxcloud_ip_address.test_v6", "network_prefix_id"),
+				),
+			},
+
+			// Test not existing IPv4 address handling
+			{
+				Config: `
+				data "anxcloud_ip_address" "test" {
+					address = "1.1.1.1"
+				}
+				`,
+				ExpectError: regexp.MustCompile("IP address was not found."),
+			},
+
+			// Test invalid IP address
+			{
+				Config: `
+				data "anxcloud_ip_address" "test" {
+					address = "1.1.1.1.1"
+				}
+				`,
+				ExpectError: regexp.MustCompile("Failed to parse IP address."),
+			},
+		},
+	})
+}
+
+func testAccAnxCloudIPAddressDataSourceResources(envInfo environment.Info) string {
```

Comment:
I like the idea, but this way we are probably creating all kinds those test IPs and prefixes for each of the tests, right?

Reply:
Only once per `TestAccAnxCloudIPAddressDataSource` testrun. The test steps are incremental.

Reply:
alright then :)

lgtm